impeller/renderer/backend/vulkan/graphics_queue_vk.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "fml/status.h"
+
+#include "impeller/base/validation.h"
+#include "impeller/renderer/backend/vulkan/command_buffer_vk.h"
+#include "impeller/renderer/backend/vulkan/command_encoder_vk.h"
+#include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "impeller/renderer/backend/vulkan/fence_waiter_vk.h"
+#include "impeller/renderer/backend/vulkan/graphics_queue_vk.h"
+#include "impeller/renderer/backend/vulkan/tracked_objects_vk.h"
+#include "impeller/renderer/command_buffer.h"
+
+namespace impeller {
+
+GraphicsQueueVK::GraphicsQueueVK(const std::weak_ptr<ContextVK>& context)
+    : context_(context) {}
+
+fml::Status GraphicsQueueVK::Submit(
+    const std::vector<std::shared_ptr<CommandBuffer>>& buffers,
+    const CompletionCallback& callback) {
+  if (buffers.empty()) {
```

Comment:
Don't know what if the "you'll always get the callback no matter what" behavior is, asked for clarification in the headerdoc. But if it is the case, you might need a success callback invocation here.

Reply:
Done